partiql-ast/src/main/java/org/partiql/ast/expr/ExprBoolTest.java

```diff
@@ -0,0 +1,45 @@
+package org.partiql.ast.expr;
+
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import org.jetbrains.annotations.NotNull;
+import org.partiql.ast.AstNode;
+import org.partiql.ast.AstVisitor;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * TODO docs
+ * Corresponds to SQL99's boolean test (6.30).
+ */
+@Builder(builderClassName = "Builder")
+@EqualsAndHashCode(callSuper = false)
+public class ExprBoolTest extends Expr {
```

Comment:
(self-review) there are a few different ways to model bool test predicate and null/missing predicate. Open to some other suggestions on modeling if it could be improved. Some alternatives I considered
- separate nodes for all (i.e. ExprIsTrue, ExprIsFalse, ExprIsUnknown, ExprIsNull, ExprIsMissing)? <- seemed a bit redundant w/ all the visitors + rewriter methods that would get added
- modeling `IS [NOT] NULL` and `IS [NOT] MISSING` together as an absent node? <- creates the same number of classes (i.e. ExprIsAbsent and AbsentEnum) as just having them separated out. I feel like we won't be adding more absent predicates? But if we will, then an enum might be better.